src/views/virtualmachines/migrate/utils.ts

```diff
@@ -0,0 +1,158 @@
+import { VirtualMachineModel } from 'src/views/dashboard-extensions/utils';
+
+import DataVolumeModel from '@kubevirt-ui/kubevirt-api/console/models/DataVolumeModel';
+import {
+  V1beta1DataVolume,
+  V1beta1StorageSpecVolumeModeEnum,
+} from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { IoK8sApiCoreV1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { MAX_NAME_LENGTH } from '@kubevirt-utils/components/SSHSecretModal/utils/constants';
+import { getName, getNamespace } from '@kubevirt-utils/resources/shared';
+import { getDataVolumeTemplates, getVolumes } from '@kubevirt-utils/resources/vm';
+import { getRandomChars } from '@kubevirt-utils/utils/utils';
+import { k8sCreate, k8sDelete, k8sPatch, Patch } from '@openshift-console/dynamic-plugin-sdk';
+
+const getBlankDataVolume = (
+  originName: string,
+  namespace: string,
+  storageClassName: string,
+  storage: string,
+): V1beta1DataVolume => ({
+  apiVersion: `${DataVolumeModel.apiGroup}/${DataVolumeModel.apiVersion}`,
+  kind: DataVolumeModel.kind,
+  metadata: {
+    name: `clone-${getRandomChars()}-${originName}`.substring(0, MAX_NAME_LENGTH),
+    namespace,
+  },
+  spec: {
+    source: {
+      blank: {},
+    },
+    storage: {
+      accessModes: ['ReadWriteMany'],
+      resources: {
+        requests: {
+          storage,
+        },
+      },
+      storageClassName,
+      volumeMode: V1beta1StorageSpecVolumeModeEnum.Block,
+    },
+  },
+});
+
+const createBlankDataVolumes = async (
+  originPVCs: IoK8sApiCoreV1PersistentVolumeClaim[],
+  destinationStorageClass: string,
+): Promise<Map<string, V1beta1DataVolume>> => {
+  const dataVolumesCreationResults = await Promise.allSettled(
+    originPVCs
+      .map((pvc) => {
+        return getBlankDataVolume(
+          getName(pvc),
+          getNamespace(pvc),
+          destinationStorageClass,
+          pvc?.spec?.resources?.requests?.storage,
+        );
+      })
+      .map((dataVolume) =>
+        k8sCreate({ data: dataVolume, model: DataVolumeModel, ns: getNamespace(dataVolume) }),
+      ),
+  );
+
+  const rejectedRequest = dataVolumesCreationResults.find((result) => result.status === 'rejected');
+
+  const fulfilledRequests = dataVolumesCreationResults.filter(
+    (result) => result.status === 'fulfilled',
+  ) as PromiseFulfilledResult<V1beta1DataVolume>[];
+
+  if (rejectedRequest) {
+    fulfilledRequests.map((result) =>
+      k8sDelete({
```

Comment:
what about failed to delete here?

Reply:
What can i do if the delete is failed? I tried :D